nelc/frontend-app-learning
# feat: supports paragon alpha and css variables

## Description
This includes the following commits: [issue](https://edunext.atlassian.net/browse/FUTUREX-979) 
1. This is a cherry pick  of this pr https://github.com/edunext/frontend-app-learning/pull/17
2. This is a fix that replaces some break conditions with css variables, taken from this commit https://github.com/eduNEXT/frontend-app-learning/pull/19/commits/712b9fe79931b90b9132662af4fe766a81c439aa, this includes an extra replacement since the edunext version doesn't include the last redwood commit which adds a new breakpoint

## How to test
1. This changes don't include default values therefore the configuration requires to specify the css file, so include this into your mfe_config variable

```
  "PARAGON_THEME_URLS": {
    "core": {
        "url": "https://cdn.jsdelivr.net/combine/npm/@edx/paragon@22.0.0-alpha.13/styles/css/themes/light/utility-classes.min.css,npm/@edx/paragon@22.0.0-alpha.13/dist/core.min.css"
    },
    "defaults": {
        "light": "light"
    },
    "variants": {
        "light": {
            "url": "https://css-varsify.s3.amazonaws.com/public/a9959998-0bab-4447-ada5-6819866195f3.css"
        }
    }
  },
```
2. delete you node module folder and run `npm install ` (USE node  18.17.0)
3.  Go to any course page a verify

## Before
![image](https://github.com/user-attachments/assets/67a5a408-5dbb-4a47-b685-5602ef846d91)

## After
![image](https://github.com/user-attachments/assets/dad7b82c-a6ed-4cc5-9133-06d74a6ce9ce)
